Fixed Conduit and Beacon Activation on Vanilla Servers

DaemonUmbra opened the pull request:
~~This solution uses DistExecutor so a better solution might be preferred, however this change allows Vanilla logic to be invoked when a Forge client connects to a Vanilla server and has its Forge tags wiped.~~
Update: As usual, Lex is right, I'm starting to see why using tags for game logic is not a good idea.

DaemonUmbra committed:
Fixed Conduit and Beacon Activation on Vanilla Servers

This solution uses DistExecutor so a better solution might be preferred, however this change allows Vanilla logic to be invoked when a Forge client connects to a Vanilla server and has its Forge tags wiped.

LexManos commented:
And this is why I dont like using tags for game logic...
This needs a cleaner solution, RIP out the tag and replace it with the normal logic.

pupnewfster commented:
Not sure this would be a good solution, and it wouldn't allow for removing the defaults, but in my opinion if you are set on using tags (given you can override it in code anyways), you could just only allow adding to defaults via tags, and have the vanilla ones hard coded. Though personally I don't really see a reason to make either of those things defined via tags.

DaemonUmbra committed:
Nixed the idea of using tags for Beacons and Conduits in light of the Vanilla-Compat issues.

Retained the addition of isConduitFrame to IForgeBlock from the previous commit, so even if pack devs can't add new Conduit Frame blocks, modders can.

DaemonUmbra committed:
Fixed Whitespace

DaemonUmbra commented:
Updated Summary:
* Supports_Beacon and Supports_Conduit tags are Nixed
* isConduitFrame added to IForgeBlockState and IForgeBlock
* ConduitTileEntity patched to use isConduitFrame
